tests/runtime/servicers/test_global_train_servicer_impl.py

```diff
@@ -40,12 +40,11 @@
 
 
 @pytest.fixture(autouse=True, params=[True, False])
-def set_train_location(request):
+def set_train_location(request, sample_train_servicer):
     """This fixture ensures that all tests in this file will be run with both
     subprocess and local training styles
     """
-    with temp_config({"training": {"use_subprocess": request.param}}):
-        yield
+    sample_train_servicer.use_subprocess = request.param
```

Comment:
I think this is still mutating global state. I think the right way to do this is:

```py
prev_value = sample_train_servicer.use_subprocess
sample_train_servicer.use_subprocess = request.param
yield
sample_train_servicer.use_subprocess = prev_value
```